install/ui/test/topology_tests.js

```diff
@@ -0,0 +1,147 @@
+        return function() {
+
+var container;
+
+QUnit.module('topology', {
+    beforeEach: function(assert) {
+        container = $('<div id="content"/>').appendTo(document.body);
+    },
+
+    afterEach: function() {
+        container.remove();
+    }
+});
+
+QUnit.test('Testing TopoGraph constructor', function(assert) {
+    var nodes = [
+        { id: 'master.ipa.test' },
+        { id: 'replica.ipa.test' }
+    ];
+
+    var graph = new topology_graph.TopoGraph({
+        nodes: nodes,
+        links: [],
+        suffixes: [
+            { cn: ['ca'] },
+            { cn: ['domain'] }
+        ]
+    });
+
+    graph.initialize(container);
+
+    assert.ok($('circle.node', container).length === nodes.length,
+        'Checking rendered nodes count');
+
+    assert.ok($('text.id:eq(0)', container).text() === 'master',
+        'Checking "master" node label');
+    assert.ok($('text.id:eq(1)', container).text() === 'replica',
+        'Checking "replica" node label');
+
+    assert.ok($('text.suffix:eq(0)', container).text() === 'ca',
+        'Checking "ca" suffix');
+    assert.ok($('text.suffix:eq(1)', container).text() === 'domain',
+        'Checking "domain" suffix');
+});
+
+QUnit.test('Testing TopoGraph links', function(assert) {
+    var nodes = [
+        { id: 'master.ipa.test', targets: {'replica.ipa.test': []} },
+        { id: 'replica.ipa.test' }
+    ];
+
+    var suffixes = [
+        { cn: ['ca'] },
+        { cn: ['domain'] }
+    ];
+
+    var links = [{
+        source: 0,
+        target: 1,
+        left: false,
+        right: true,
+        suffix: suffixes[0]
+    }];
+
+    var graph = new topology_graph.TopoGraph({
+        nodes: nodes,
+        links: links,
+        suffixes: suffixes
+    });
+
+    graph.initialize(container);
+
+    assert.ok($('circle.node', container).length === nodes.length,
+        'Checking rendered nodes count');
+
+    var rendered_links = $('path.link', container).not('.dragline');
+    assert.ok(rendered_links.length === 1,
+        'Checking right direction link is rendered');
+
+    var marker = rendered_links.first().css('marker-end');
+    assert.ok(marker && marker !== 'none',
+        'Checking right direction link has proper marker');
+
+    links.push({
+        source: 0,
+        target: 1,
+        left: true,
+        right: false,
+        suffix: suffixes[1]
+    })
+    graph.update(nodes, links, suffixes);
+
+    rendered_links = $('path.link', container).not('.dragline')
+    assert.ok(rendered_links.length === 2,
+        'Checking left direction link is rendered');
+
+    marker = rendered_links.last().css('marker-start');
+    assert.ok(marker && marker !== 'none',
+        'Checking left direction link has proper marker');
+});
+
+QUnit.test('Testing TopoGraph for multiple DNS zones', function(assert) {
+    var nodes = [
+        { id: 'master.zone1' },
+        { id: 'replica.zone1' },
+        { id: 'master.zone2' },
```

Comment:
FreeIPA does not allow single label domains, you could use instead master.sub.zone1 replica.sub.zone1, master.sub.zone2.

Reply:
Fixed